stack/stack_test.go

```diff
@@ -48,175 +49,235 @@ const TestData_2 string = `provider:
 
 `
 
+const noMatchesErrorMsg string = "No functions matching --filter/--regex were found in the YAML file"
+const invalidRegexErrorMsg string = "error parsing regexp"
+
 var ParseYAMLTests_Regex = []struct {
-	title       string
-	searchTerm  string
-	functions   []string
-	file        string
-	expectError bool
+	title            string
+	searchTerm       string
+	functions        []string
+	file             string
+	expectError      bool
+	expectedErrorMsg string
 }{
 	{
-		title:       "Regex search for functions only containing 'node'",
-		searchTerm:  "node",
-		functions:   []string{"nodejs-echo"},
-		file:        TestData_1,
-		expectError: false,
+		title:            "Regex search for functions only containing 'node'",
+		searchTerm:       "node",
+		functions:        []string{"nodejs-echo"},
+		file:             TestData_1,
+		expectError:      false,
```

Comment:
Could get rid of the `expectedErrorMsg` and change `expectError` to a string, use `len(test.expectError) > 0` for the conditional - avoids having the redundant variable since you're always checking for the message.

Reply:
Thanks John, will update this evening.